src/coord/src/catalog.rs

```diff
@@ -193,12 +193,22 @@ pub struct Index {
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum Type {
     Map {
+        create_sql: String,
+        plan_cx: PlanContext,
         oid: u32,
-        key_oid: u32,
-        value_oid: u32,
+        key_ids: TypeIds,
+        value_ids: TypeIds,
     },
 }
 
+/// All data types available for use in Materialize will
+/// have an OID. Only user-defined types will have a GlobalId.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TypeIds {
+    pub id: Option<GlobalId>,
+    pub oid: u32,
+}
```

Comment:
It occurs to me that I think we should enforce that all system types have a global ID, too. Specifically by adding a new `Builtin::Type` thing in coord/src/catalog/builtin.rs, and then loading that up into the catalog at boot. That'll help along the npgsql work! And then you won't need to do anything special for type IDs here; just `key_id: GlobalId, value_id: GlobalId`.